### PR TITLE
docs: update .readthedocs.yaml

### PR DESCRIPTION
**Pull request by robrap**
**Description:**

Following cookiecutter recommendations in this update: https://github.com/openedx/edx-drf-extensions/commit/bb2b78347d96145d56e5a3b671ad36e604be612e

- Rename .readthedocs.yml to .readthedocs.yaml
- Remove extra python install lines from edx-platform
- Add fail_on_warning to protect against doc issues

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Commits are squashed

### CODE REVIEW

**Comment by feanil** [comment]
It looks like this repo does need to install `.` because it's making use of the `automodule` feature for documentation.  Most of our repos aren't using this and not all repos are pip installable so it might not make sense as a part of the default but also it makes sense here.

File: .readthedocs.yaml [hunk 1]
```diff
@@ -13,9 +13,8 @@ build:
 # Build documentation in the docs/ directory with Sphinx
 sphinx:
   configuration: docs/conf.py
+  fail_on_warning: true
 
 python:
   install:
   - requirements: "requirements/docs.txt"
```
Comment (robrap) [inline .readthedocs.yaml:20]:
```suggestion
  - requirements: "requirements/docs.txt"
  - method: pip
    path: .
```

**Comment by robrap** [comment]
@feanil: Any interest in fixing up `fail_on_warning` issues? Otherwise, I'm going to close this PR and move on to other things. Not sure that the file rename really matters.

**Comment by feanil** [comment]
yea, I'll take a look

**Review by feanil — APPROVED** [review]

**Comment by feanil** [comment]
This is fixed now, just needs codcov to recieve the coverage info and then it should be good to merge.

**Comment by robrap** [comment]
@feanil: Thank you!

**Pull request merged**